src/components/Bindings.tsx

```diff
@@ -40,7 +32,7 @@ const Bindings = (props: {
     NonEmptyArray<Readonly<NonEmptyArray<UIKeyDescriptor>>>
   > = [
     [
-      { key: "`", action: props.keyMap["`"] }, // this is on its own line so I don't need to fight with typescript for it to know that the array is NonEmptyArray
+      { key: "`", action: props.keyMap["`"], width: "3em" }, // make typing work
```

Comment:
Once you do something other than just `{ key, action }`, you're not isolating it from `...Keys(["1", "2", "3", "4", "5", "6", "7", "8", "9", "0"])` for the purpose of typing. It's separate because it has to be, and thus the comment is now useless.

Reply:
so this was something i was confused about. i think the typing is currently… too strict?

Reply:
it's not too strict, actually the `.map` operator has typings too weak. namely, it doesn't know that a non-empty list (`NonEmptyArray<T>` type, which is just `[T, ...T]`) gets mapped to a non-empty list.

by keeping the ` key out of the map, i was directly telling TS that the array is non-empty. this doesn't matter anymore, since you need to keep it out of the map anyways (to set the width)